### PR TITLE
Move more intrinsics processing logic to IntrinsicGenerator

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt [hunk 1]
```diff
@@ -54,48 +83,299 @@ internal class IntrinsicGenerator(val codegen: CodeGenerator) {
...
+                IntrinsicType.LIST_OF_INTERNAL -> emitListOfInternal(callSite, args)
+                IntrinsicType.IDENTITY -> emitIdentity(args)
+                IntrinsicType.GET_CONTINUATION -> emitGetContinuation()
+            }
+
+    private fun FunctionGenerationContext.emitGetContinuation(): LLVMValueRef =
+            continuationProvider()
+
+    private fun FunctionGenerationContext.emitIdentity(args: List<LLVMValueRef>): LLVMValueRef =
+            args[0]
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt]: Why not `single`?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt [hunk 2]
```diff
@@ -54,48 +83,299 @@ internal class IntrinsicGenerator(val codegen: CodeGenerator) {
...
+        return ifThenElse(superClassIsNull, receiver) {
+            val structType = structType(kInt8Ptr, kInt8Ptr)
+            val ptr = alloca(structType)
+            store(receiver, LLVMBuildGEP(builder, ptr, cValuesOf(kImmZero, kImmZero), 2, "")!!)
+            store(superClass, LLVMBuildGEP(builder, ptr, cValuesOf(kImmZero, kImmOne), 2, "")!!)
+            bitcast(int8TypePtr, ptr)
+        }
+    }
+
+    // TODO: Find better place for these guys
```
Comment (olonho) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt]: Please dot at the end of comment.

File: shared/src/library/kotlin/org/jetbrains/kotlin/konan/library/SearchPathResolver.kt [hunk 3]
```diff
@@ -203,7 +203,7 @@ internal fun SearchPathResolverWithTarget.libraryMatch(candidate: KonanLibraryIm
     val candidateAbiVersion = candidate.versions.abiVersion
     val candidateLibraryVersion = candidate.versions.libraryVersion
 
-    if (resolverTarget != null && !candidate.targetList.contains(resolverTarget.visibleName)) {
```
Comment (olonho) [inline shared/src/library/kotlin/org/jetbrains/kotlin/konan/library/SearchPathResolver.kt]: Why this change?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt (hunk 1 above)
Comment (sbogolepov) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt]: Forgot about this one.

File: shared/src/library/kotlin/org/jetbrains/kotlin/konan/library/SearchPathResolver.kt (hunk 3 above)
Comment (sbogolepov) [reply shared/src/library/kotlin/org/jetbrains/kotlin/konan/library/SearchPathResolver.kt]: Found that it's always true when was fixing failing test.
